src/NuGet.Core/NuGet.Configuration/PackageSourceMapping/PackageSourceMappingProvider.cs

```diff
@@ -58,5 +59,41 @@ public void AddOrUpdatePackageSourceMappingSourceItem(PackageSourceMappingSource
 
             _settings.SaveToDisk();
         }
+
+        public void SavePackageSourceMappings(IReadOnlyList<PackageSourceMappingSourceItem> packageSourceMappingsSourceItems)
+        {
+            if (packageSourceMappingsSourceItems == null)
+            {
+                throw new ArgumentNullException(nameof(packageSourceMappingsSourceItems));
+            }
+
+
+            var existingSettingsLookup = GetPackageSourceMappingItems();
+
+            foreach (var sourceMappingItem in packageSourceMappingsSourceItems)
+            {
+                if (existingSettingsLookup != null)
+                {
+                    AddOrUpdatePackageSourceMappingSourceItem(sourceMappingItem);
+                }
+            }
+
+            //Remove all old mappings not in new mappings
+            if (existingSettingsLookup != null)
+            {
+                ObservableCollection<PackageSourceMappingSourceItem> removeMappings = new ObservableCollection<PackageSourceMappingSourceItem>();
```

Comment:
Do we really need `ObservableCollection` here? I believe just `List` should work. ObservableCollection does more things, more heavy.

Reply:
Changed the `ObservableCollections` to `Lists`. I was confused about the difference between them when I originally wrote the code